src/shared/bus-unit-util.c

```diff
@@ -561,9 +561,9 @@ static int bus_append_socket_filter(sd_bus_message *m, const char *field, const
 static int bus_append_exec_command(sd_bus_message *m, const char *field, const char *eq) {
         bool explicit_path = false, done = false, ambient_hack = false;
         _cleanup_strv_free_ char **cmdline = NULL, **ex_opts = NULL;
-        _cleanup_free_ char *_path = NULL, *upgraded_name = NULL;
+        _cleanup_free_ char *_path = NULL;
         ExecCommandFlags flags = 0;
-        bool is_ex_prop = endswith(field, "Ex");
+        bool ex_prop = false;
```

Comment:
Maybe leave it uninitialized here, rather set this to false in the if branch where Ex is chopped?

Reply:
I rearranged the code to just set `ex_prop` to the right value once.